pkg/engine/engine.go

```diff
@@ -108,12 +176,101 @@ func filterDetectors(filterFunc func(detectors.Detector) bool, input []detectors
 	return output
 }
 
-func Start(ctx context.Context, options ...EngineOption) *Engine {
+func (e *Engine) setFoundResults() {
+	atomic.StoreUint32(&e.foundResults, 1)
+}
+
+// HasFoundResults returns true if any results are found.
+func (e *Engine) HasFoundResults() bool {
+	return atomic.LoadUint32(&e.foundResults) == 1
+}
+
+// GetMetrics returns a copy of Metrics.
+// It's safe for concurrent use, and the caller can't modify the original data.
+func (e *Engine) GetMetrics() Metrics {
+	e.metrics.mu.RLock()
+	defer e.metrics.mu.RUnlock()
+
+	result := Metrics{
+		BytesScanned:           e.metrics.BytesScanned,
+		ChunksScanned:          e.metrics.ChunksScanned,
+		VerifiedSecretsFound:   e.metrics.VerifiedSecretsFound,
+		UnverifiedSecretsFound: e.metrics.UnverifiedSecretsFound,
+		AvgDetectorTime:        map[string]time.Duration{},
+	}
```

Comment:
**Nit:** since `e.metrics` is an object (not a pointer), we can initialize it like so:

```go
result := e.metrics.Metrics
result.AvgDetectorTime = make(map[string]time.Duration, len(e.metrics.AvgDetectorTime))
```

This will catch any added attributes without having to update this function.